### PR TITLE
Add test coverage, etc.

### PR DESCRIPTION
**Pull request by vijaypemmaraju**
Big diff, but mostly because i decided to move off of yarn.

The main purpose of this PR is to add test coverage so that we have at least one repo that has 100% test coverage.

In addition, I've also done these things:
- add eslint groove config
- switch from yarn to npm
- improve our travis build setup to be similar to other repos.
- add code climate coverage

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,20 +1,24 @@
 interface StringMap { [key: string]: string | StringMap }
 
 function createActionsHelper(actionKey: string, actions: string | StringMap): string | StringMap {
-  if (typeof actions === 'string') {
-    if (actionKey) {
-      return `${actionKey}/${actions}`;
-    }
-
-    return actions;
```
Comment (vijaypemmaraju) [inline src/index.ts:9]: removing this return was the one "real" implementation change I made. I don't think this line actually ever gets hit. I found this out by looking through coverage and seeing that this line was always uncovered.

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
+  - name: Release
+    if: branch = master AND type != pull_request
+
 jobs:
   include:
+    - stage: Prepare
+      name: Test
+      before_script:
+        - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+        - chmod +x ./cc-test-reporter
```
Comment (corytheboyd) [inline .travis.yml:20]: shouldn't need to modify file permissions if we check that file in with them

Comment (vijaypemmaraju) [reply .travis.yml:20]: this is recommended by [their docs](https://docs.codeclimate.com/docs/travis-ci-test-coverage) 🤷

Comment (corytheboyd) [reply .travis.yml:20]: weird lol, it's not a big deal so I'm fine leaving it

**Pull request merged**